command/agent/auth/azure/azure.go

```diff
@@ -30,6 +30,8 @@ type azureMethod struct {
 
 	role     string
 	resource string
+	objectID string
```

Comment:
Not for this PR, but it would be nice to switch auto_auth config to use `hcl` tags more consistently.